### PR TITLE
Hypergraphs and multi-layers networks

### PR DESCRIPTION
**Pull request by tpoisot**

### CODE REVIEW

**Comment by tpoisot** [comment]
@Norma-Forero I've made a first draft of this answer, your turn!

**Comment by tpoisot** [comment]
@Norma-Forero do you want to make a first attempt at this answer? I can do the multi-layer part next.

**Comment by tpoisot** [comment]
@Norma-Forero I tried to add some references, but I think I am missing a few on hypergraphs - can you read through it and make the changes? I think this paragraph is almost ready for review.

**Comment by gottacatchenall** [comment]
hey folks--- curious if you had seen [this paper on stacking models link-prediction](https://arxiv.org/abs/1909.07578). one of the points they make in the discussion is that the link-prediction algorithms commonly used/applied in social network contexts don't work as well in biological networks, may be relevant to that sentence

**Comment by tpoisot** [comment]
@gottacatchenall I've read that when it came out - not sure it specifically applies here (but should def. be cited somewhere else in the MS). We also show in Becker et al. 2020 that ensemble models based on structure, ecology, and mechanisms Just Work :tm:

**Comment by tpoisot** [comment]
@TanyaS08 & @gottacatchenall - this one is ready for review and merging into the intermediate branch

**Review by gottacatchenall — COMMENTED** [review]
@tpoisot @TanyaS08 more of a general question: do we want to save minor copyediting things for the integration -> master PR?

**Comment by tpoisot** [comment]
> @tpoisot @TanyaS08 more of a general question: do we want to save minor copyediting things for the integration -> master PR?

I think it's probably more efficient to make the changes as part of this branch.

**Comment by tpoisot** [comment]
We might want to cite

> Gross, Thilo, Korinna T. Allhoff, Bernd Blasius, Ulrich Brose, Barbara Drossel, Ashkaan K. Fahimipour, Christian Guill, Justin D. Yeakel, and Fanqi Zeng. “Modern Models of Trophic Meta-Communities.” Philosophical Transactions of the Royal Society B: Biological Sciences 375, no. 1814 (December 21, 2020): 20190455. https://doi.org/10.1098/rstb.2019.0455.

@Norma-Forero what do you think, :+1: or :-1:? It's in the zotero group as `Gross2020ModMod` already.

File: manuscript.md [hunk 1]
```diff
@@ -42,7 +46,35 @@
...
-## Could we use hypergraphs and multi-layer networks to predict more interactions? 
+## Could we use hypergraphs and multi-layer networks to predict more interactions?
+
+The web of life is built by a variety of interactions that can vary over time
+and space; these interactions include facilitation, niche construction,
+zoonoses, vector-borne diseases, among others [@Garcia-Callejas2018MulInt],
+which all share the fact that interactions between species are themselves
+interacting. One mathematical tool to describe these situations is hypergraphs:
+hypergraphs are the generalization of a graph, allowing a broad and manageable
+approach to complex interactions [@Carletti2020DynSys]. @Golubski2011ModMod were
```
Comment (TanyaS08) [inline manuscript.md:57]: I wonder id this one statement on what a hypergdaph is is sufficient. You are pointing the reader to a further reading opportunity so it probably is okay. Also after writing this and then checking the suggested glossary I see it is flagged there so I guess this is all good then - but maybe taking a step back in the glossary and expanding on the idea/concept of networks as graphs? Not sure if that comes up elsewhere in the ms....

File: manuscript.md [hunk 2]
```diff
@@ -42,7 +46,35 @@
...
+among the first to show that interaction modifiers are themselves interacting,
+which makes the complexity of ecological networks explode. Investigating
+hypergraphical interactions can be more important than previously thought, as
+they can reveal the importance of species within a network based on *indirect*
+interactions [@Golubski2016EcoNet]. An additional degree of complexity is
+introduced by multi-layer networks [@Hutchinson2019SeeFor]. Multi-layer networks
+offer links across "variants" of the networks, which can be patches, timepoints,
+environments, etc. As @Pilosof2017MulNat suggest, ecological networks are
+intrinsically multi-layers, as they are shaped by evolution, dispersal,
```
Comment (TanyaS08) [inline manuscript.md:66]: I'm not to clued up on terminology not sure if grammar errors or not - should it maybe be: 'networks are intrinsically *multi-layered*'?

**Review by TanyaS08 — APPROVED** [review]
flagged to very minor points (the first is more of a mental ramble than anything) - as is I think it's ready to go to the integration branch but if you make further changes just tag me in for a second look 😁

File: manuscript.md [hunk 3]
```diff
@@ -42,7 +46,35 @@
 
 ## How are interaction strengths actually inferred? 
 
-## Could we use hypergraphs and multi-layer networks to predict more interactions? 
+## Could we use hypergraphs and multi-layer networks to predict more interactions?
```
Comment (gottacatchenall) [inline manuscript.md:49]: > predict more interactions?

i think the wording of the question here could be sharpened---does "more interactions" mean improving prediction accuracy of interactions?

File: manuscript.md [hunk 4]
```diff
@@ -42,7 +46,35 @@
...
+
+The web of life is built by a variety of interactions that can vary over time
+and space; these interactions include facilitation, niche construction,
+zoonoses, vector-borne diseases, among others [@Garcia-Callejas2018MulInt],
+which all share the fact that interactions between species are themselves
```
Comment (gottacatchenall) [inline manuscript.md:54]: >  which all share the fact that interactions between species are themselves are interacting

i think we should be more clear the hypergraphs are what we are using in the higher-order interaction case

and that multilayer networks are distinct from hypergraphs and used for 
- nodes are species, layers are different time points/locations (the pilosof paper referenced later) 
- nodes are spatial coordinates, layers represent dispersal between species ([example 1](https://conbio.onlinelibrary.wiley.com/doi/pdf/10.1111/cobi.12943), [example 2](https://besjournals-onlinelibrary-wiley-com.proxy3.library.mcgill.ca/doi/pdf/10.1111/1365-2435.13237))

File: manuscript.md (hunk 1 above)
Comment (gottacatchenall) [reply manuscript.md:57]: i think it needs to be more clear that a hypergraph considers edges between more than 2 nodes

File: manuscript.md (hunk 2 above)
Comment (gottacatchenall) [reply manuscript.md:66]: i'm also fond of [this paper](https://conbio.onlinelibrary.wiley.com/doi/pdf/10.1111/cobi.12943) if we have room for an extra citation

**Review by gottacatchenall — CHANGES_REQUESTED** [review]

File: manuscript.md (hunk 1 above)
Comment (tpoisot) [reply manuscript.md:57]: This is a good call, I will fix it in the next commit.

File: manuscript.md (hunk 3 above)
Comment (tpoisot) [reply manuscript.md:49]: I flipped the question a little bit to reflect the fact that we might want to predict these types of networks as well.

File: manuscript.md (hunk 4 above)
Comment (tpoisot) [reply manuscript.md:54]: I have re-organized the answer into two paragraphs, to make this clearer.

File: manuscript.md (hunk 2 above)
Comment (tpoisot) [reply manuscript.md:66]: Both fixed in the next commit.

**Review by gottacatchenall — APPROVED** [review]
:+1:

**Pull request merged**